### DISCUSSION TITLE
How to get the mixture fraction diffusion coefficient using PeleC?

### DISCUSSION THREAD

**Discussion by Rachel008996** [AMReX-Combustion/PeleC#768] [Q&A] [answered]
![image](https://github.com/AMReX-Combustion/PeleC/assets/118787120/b12f96a9-0d19-4330-90a6-3963eef4c39b)

**Comment by hsitaram** [comment]
Hi Rachel, in the cavity flame paper our mixture fraction was defined based on Y_N2. So we used yt to get covering grid data and then used Cantera to extract the N2's diffusion coefficient. Now I think, PeleC, has the Ds in the plot files as well.

**Reply by Rachel008996** [reply]
Hello, hsitaram! Thank you very much for your reply. I couldn't find Ds in the output plt files. However, I found a post-processing program named plotTransportCoeff.cpp under PeleAnalysis/Src/ModelSpecificAnalysis, which I think might be useful. Since I am not very familiar with Pele, I wonder if the D_a obtained through 'trans.get_transport_coeffs(tbx, Y_a, T_a, rho_a, D_a, mu_a, xi_a, lam_a, ltransparm)' is the D I am looking for (in the output plt files, it appears in forms like rhoD(CO)). Thank you very much for considering my question. I wish you a pleasant day!

**Reply by marchdf** [reply]
I was pretty sure there was a way to get `D`s... @baperry2 do you remember? if you try `amr.derive_plot_vars=ALL` it will plot _all_ derived variables. It might show up in that list. It's probably more than you want...

**Reply by drummerdoc** [reply]
PeleLMeX has an input called "run_mode" which can be set to "normal" (default) or "evaluate" where it restarts by reading in all the stuff from a checkpoint or initialization, then evaluates anything requested in the derive_plot_vars list.  If it wasn't there to start with (I thought it was) evaluating the transport coefficients would be an easy add.  If this sounds useful, we can put it on a list somewhere of things to add (by copying over code from LMeX).

**Reply by baperry2** [reply, marked as answer]
To dump diffusivities in plotfiles at runtime in PeleC, use `amr.derive_plot_vars = diffusivity conductivity viscosity`. The diffusivities this outputs are labeled `D(species)` but I'm pretty sure they are actually `rhoD(species)`. Dumping your desired outputs at runtime is the preferred approach.

As @drummerdoc mentions, PeleLMeX run mode might also be helpful. See the discussion [here](
https://github.com/AMReX-Combustion/PeleLMeX/discussions/301#discussioncomment-7490958). It's possible to restart PeleLMeX from a PeleC plot file (`amr.initDataPlt = plt00200` and `amr.initDataPltSource = C`), so theoretically you could use the PeleLMeX evaluate mode to compute derived variables from a PeleC simulation.

I haven't used the PeleAnalysis tool that you are using, but it looks like it should also give you the result you want (after dividing by rho to get D from rhoD).
